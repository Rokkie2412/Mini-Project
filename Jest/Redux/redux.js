export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer