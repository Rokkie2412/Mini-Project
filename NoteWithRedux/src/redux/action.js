import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})