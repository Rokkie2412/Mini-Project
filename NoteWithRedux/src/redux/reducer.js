import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}