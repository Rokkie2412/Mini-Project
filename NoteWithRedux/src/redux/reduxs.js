import remove from "lodash.remove"
export const ADDARRAY = 'ADDARRAY'
export const DELETENOTE = 'DELETENOTE'
export const EDITNOTE = 'EDITNOTE'

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})

export const deleteFromArray = (title) => ({
    type:DELETENOTE,
    payload:title
})
export const EditNote = (title) => ({
    type:EDITNOTE,
    payload:title,
})



const initialState = []

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return[
                ...state,action.payload
            ]
        case DELETENOTE:
            const deleteNewArray = remove(state, obj => {
                return obj.title != action.payload
            })
            return deleteNewArray
        case EDITNOTE:
            // const filterDelete = map(state,obj =>{
            //     return obj.title == action.payload
            // })
            const filtered = state.filter((state,obj)=>{
                if(obj.title === action.payload){
                    var tamp = obj.title
                    console.log(tamp);
                }
            })
        default:
            return state;
    }
}

export default mainReducer