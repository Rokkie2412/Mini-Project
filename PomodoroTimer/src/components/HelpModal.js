import { KeyboardAvoidingView,View,Text,Pressable,Modal } from 'react-native';
import React from 'react';
import styles from '../styles/HelpModalStyle'
import Ion from 'react-native-vector-icons/Ionicons'
const HelpModal = ({HelpVisible,setHelpVisible}) =>{
    return(
        <Modal
        visible={HelpVisible}
        animationType='fade'
        transparent={true}
        >
            <KeyboardAvoidingView style={styles.mainContainer}>
                <View style={styles.CloseContainer}>
                    <Pressable onPress={()=>setHelpVisible(false)}><Ion name='close' style={styles.closeIcon}/></Pressable>
                </View>
                <Text style={styles.Header}>How to use</Text>
                <View style={styles.TextContainer}>
                    <View style={styles.textStruct}>
                        <Text style={styles.text1}>1.</Text>
                        <Text style={styles.text2}>For better convenience, please lock the app on background</Text>
                    </View>
                    <View style={styles.textStruct}>
                        <Text style={styles.text1}>2.</Text>
                        <Text style={styles.text2}>Before use Doro Timer, you can set your timer that set for you in setting button</Text>
                    </View>
                    <View style={styles.textStruct}>
                        <Text style={styles.text1}>3.</Text>
                        <Text style={styles.text2}>To start timer, press play button, and to pause press pause button. Button will change every you press the button</Text>
                    </View>
                    <View style={styles.textStruct}>
                        <Text style={styles.text1}>4.</Text>
                        <Text style={styles.text2}>To stop timer, press stop button, timer will get reset to first condition</Text>
                    </View>
                    <View style={styles.textStruct}>
                        <Text style={styles.text1}>5.</Text>
                        <Text style={styles.text2}>When timer on you can't open setting, you must stop your timer first</Text>
                    </View>
                </View>
                <Pressable
                onPress={()=>setHelpVisible(false)}
                 style={styles.CloseSction}>
                    <Text style={styles.close}>Close</Text>
                </Pressable>
            </KeyboardAvoidingView>
        </Modal>
    )
}

export default HelpModal