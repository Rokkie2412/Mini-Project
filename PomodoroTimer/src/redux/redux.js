
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer