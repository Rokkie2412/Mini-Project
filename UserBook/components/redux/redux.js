export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}