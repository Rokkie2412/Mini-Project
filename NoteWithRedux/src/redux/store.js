import { createStore } from "redux";
import {mainReducer} from "./reduxs";

export const store = createStore(mainReducer)