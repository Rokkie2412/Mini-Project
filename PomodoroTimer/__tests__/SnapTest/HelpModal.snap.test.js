import React from 'react';
import {create,act} from 'react-test-renderer';
import HelpModal from '../../src/components/HelpModal';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {mainReducer} from '../../src/redux/redux';

jest.mock('react-native-vector-icons/Ionicons', () => 'MockedIonicons');

jest.mock('react-native-background-timer', () => 'Mocked Background Timer');

jest.mock('react-native-push-notification', () => 'Mocked Push Notification');

jest.useFakeTimers();
const setHelpVisibleMock = jest.fn()
const HelpVisibleMock = jest.fn()
const store = createStore(mainReducer, {status: 'default status'});
const tree = create(
  <Provider store={store}>
    <HelpModal setHelpVisible={setHelpVisibleMock}/>
  </Provider>,
);


describe('Render HelpModal Screen',()=>{
  describe('Snapshot for help modal screen',()=>{
    it('Render correctly for Help Modal Screen', () => {
    expect(tree).toMatchSnapshot();
    });
  })
  describe("setup Help Visible Modal",()=>{
    it('Testing visible button',()=>{
      const button = tree.root.findByProps({testID:'setupVisibleModal'}).props
      act(()=>button.onPress())
    })
  })
  describe("setup Help Close Modal",()=>{
    it('Testing visible button',()=>{
      const button = tree.root.findByProps({testID:'setupVisibleModal'}).props
      act(()=>button.onPress())
    })
  })
    describe("setup Help Close Modal",()=>{
    it('Testing visible button',()=>{
      const button = tree.root.findByProps({testID:'VisibleModal'}).props
      act(()=>button.onPress())
    })
  })
})
