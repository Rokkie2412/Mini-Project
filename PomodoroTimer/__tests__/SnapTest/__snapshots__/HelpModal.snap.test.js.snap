// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HelpModal Screen Snapshot for help modal screen Render correctly for Help Modal Screen 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "#393E46",
        "borderRadius": 30,
        "flex": 1,
        "height": "75%",
        "margin": "auto",
        "marginHorizontal": "5%",
        "marginVertical": "25%",
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="setupVisibleModal"
      >
        <MockedIonicons
          name="close"
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 30,
              "margin": 10,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#EEEE",
          "fontSize": 30,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      How to use
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#222831",
          "borderRadius": 25,
          "height": "71%",
          "margin": 12,
          "paddingHorizontal": 20,
          "width": "92%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 17,
              "paddingLeft": 6,
              "paddingVertical": 17,
            }
          }
        >
          1.
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 17,
            }
          }
        >
          For better convenience, please lock the app on background
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 17,
              "paddingLeft": 6,
              "paddingVertical": 17,
            }
          }
        >
          2.
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 17,
            }
          }
        >
          Before use Doro Timer, you can set your timer that set for you in setting button
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 17,
              "paddingLeft": 6,
              "paddingVertical": 17,
            }
          }
        >
          3.
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 17,
            }
          }
        >
          To start timer, press play button, and to pause press pause button. Button will change every you press the button
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 17,
              "paddingLeft": 6,
              "paddingVertical": 17,
            }
          }
        >
          4.
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 17,
            }
          }
        >
          To stop timer, press stop button, timer will get reset to first condition
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 17,
              "paddingLeft": 6,
              "paddingVertical": 17,
            }
          }
        >
          5.
        </Text>
        <Text
          style={
            Object {
              "color": "#EEEE",
              "fontSize": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 17,
            }
          }
        >
          When timer on you can't open setting, you must stop your timer first
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
      testID="VisibleModal"
    >
      <Text
        style={
          Object {
            "backgroundColor": "#00ADB5",
            "borderRadius": 50,
            "color": "#EEEE",
            "fontSize": 22,
            "fontWeight": "bold",
            "padding": 10,
          }
        }
      >
        Close
      </Text>
    </View>
  </View>
</Modal>
`;
